hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -603,4 +611,70 @@ private static boolean isDownloadAndUpdateRequest(final DmfMultiActionElement mu
+    public void assertBatchAssignmentsDownloadAndInstall() {
+        assertBatchAssignmentsMessagePayload(BATCH_DOWNLOAD_AND_INSTALL);
+    }
+
+    @Test
+    @Description("Verify payload of batch assignments download only message.")
+    public void assertBatchAssignmentsDownloadOnly() {
+        assertBatchAssignmentsMessagePayload(BATCH_DOWNLOAD);
+    }
+
+    protected void assertBatchAssignmentsMessagePayload(final EventTopic topic) {
+        enableBatchAssignments();
+
+        final List<String> targets = Arrays.asList("batchCtrlID1", "batchCtrlID2", "batchCtrlID3");
+        for (int i = 0; i < targets.size(); i++) {
+            registerAndAssertTargetWithExistingTenant(targets.get(i), i+1);
+        }
+
+        final DistributionSet ds = testdataFactory.createDistributionSet();
+        testdataFactory.addSoftwareModuleMetadata(ds);
+
+        final DistributionSetAssignmentResult assignmentResult = assignDistributionSet(ds.getId(), targets,
+                topic == BATCH_DOWNLOAD?DOWNLOAD_ONLY:FORCED);
+
+        waitUntilEventMessagesAreDispatchedToTarget(topic);
+
+        final Message message = replyToListener.getLatestEventMessage(topic);
+        final Map<String, Object> headers = message.getMessageProperties().getHeaders();
+        assertThat(headers).containsEntry("type", EVENT.toString());
+        assertThat(headers).containsEntry("topic",topic.toString());
+
+        final DmfBatchDownloadAndUpdateRequest batchRequest = (DmfBatchDownloadAndUpdateRequest) getDmfClient()
+                .getMessageConverter().fromMessage(message);
+
+        assertThat(batchRequest).isExactlyInstanceOf(DmfBatchDownloadAndUpdateRequest.class);
+        assertDmfBatchDownloadAndUpdateRequest(batchRequest, ds.getModules(), targets);
+    }
+
+    protected void assertDmfBatchDownloadAndUpdateRequest(final DmfBatchDownloadAndUpdateRequest request,
+                                                     final Set<SoftwareModule> softwareModules,
+                                                     final List<String> controllerIds) {
+        assertThat(softwareModules)
+                .is(new HamcrestCondition<>(SoftwareModuleJsonMatcher.containsExactly(request.getSoftwareModules())));
+        request.getSoftwareModules().forEach(dmfModule -> assertThat(dmfModule.getMetadata()).containsExactly(
+                new DmfMetadata(TestdataFactory.VISIBLE_SM_MD_KEY, TestdataFactory.VISIBLE_SM_MD_VALUE)));
+
+
+        List<Object> tokens = controllerIds.stream().map(controllerId -> {
+            final Optional<Target> target = controllerManagement.getByControllerId(controllerId);
+            assertThat(target).isPresent();
+            return target.get().getSecurityToken();
+        }).collect(Collectors.toList());
+
+
+        List<Target> requestTargets = request.getTargets().stream().map(
+                dmfTarget -> waitUntilIsPresent(() -> targetManagement.getByControllerID(dmfTarget.getControllerId()))
```

Comment:
`controllerManagement.getByControllerId` and `targetManagement.getByControllerID` are effectively the same, the only difference is the required permission to execute the methods. Both of them retrieve targets from DB, thus what you effectively doing here is comparing two identical data sets. Instead, you would need to extract the controllerId->securityToken mapping from the request and compare it with the security token saved in DB for that target

Reply:
done

Reply:
the issue is still valid, could you check it? Contact me if you need more clarification